src/settings/global_user.rs

```diff
@@ -86,40 +166,97 @@ impl GlobalUser {
     }
 
     pub fn to_file(&self, config_path: &Path) -> Result<()> {
-        let toml = toml::to_string(self)?;
+        // convert in-memory representation of authentication method to on-disk format
+        let toml: std::string::String = toml::to_string(self)?;
 
+        // create configuration path if non-existent, and write to configuration file
         fs::create_dir_all(&config_path.parent().unwrap())?;
         fs::write(&config_path, toml)?;
 
         Ok(())
     }
 
     fn from_config(config: config::Config) -> Result<Self> {
-        let global_user: Result<GlobalUser, config::ConfigError> = config.clone().try_into();
-        match global_user {
-            Ok(user) => Ok(user),
-            Err(_) => {
-                let wrangler_login_msg = styles::highlight("`wrangler login`");
-                let wrangler_config_msg = styles::highlight("`wrangler config`");
-                let vars_msg = styles::url("https://developers.cloudflare.com/workers/tooling/wrangler/configuration/#using-environment-variables");
-                let msg = format!(
-                    "{} Your authentication details are improperly configured.\nPlease run {}, {}, or visit\n{}\nfor info on configuring with environment variables",
-                    emoji::WARN,
-                    wrangler_login_msg,
-                    wrangler_config_msg,
-                    vars_msg
-                );
-                log::info!("{:?}", config);
-                anyhow::bail!(msg)
-            }
+        // Get all the possible authentication methods
+        let api_token = config.get_str("api_token");
+        let oauth_token = config.get_str("oauth_token");
+        let email = config.get_str("email");
+        let api_key = config.get_str("api_key");
+        let refresh_token = config.get_str("refresh_token");
+        let expiration_time = config.get_str("expiration_time");
+
+        // The only cases that are not allowed are:
+        //      1) (partial or complete) OAuth token + API token
+        //      2) (partial or complete) OAuth token + Global API key (partial or complete)
+        //      3) Invalid authentication methods (e.g. partial Oauth token, partial Global API key, and empty configuration file + no environment variables)
+        // API token has priority over global API key both in environment variables and in configuration file
+        if (api_token.is_ok()
+            && (oauth_token.is_ok() || refresh_token.is_ok() || expiration_time.is_ok()))
+            || ((oauth_token.is_ok() || refresh_token.is_ok() || expiration_time.is_ok())
+                && (email.is_ok() || api_key.is_ok()))
+        {
+            let error_info = "\nMore than one authentication method (e.g. API token and OAuth token, or OAuth token and Global API key) has been found in the configuration file. Please use only one.";
```

Comment:
We can probably add more detail to this error:
- print the path to the config file in case they want to manually edit it
- suggest running `wrangler logout` (assuming this clears _all_ auth info on disk?)